app/factories/shopify/variant_factory.rb

```diff
@@ -0,0 +1,31 @@
+module Shopify
+  class VariantFactory
+    def initialize(spree_variant, shopify_variant)
+      @spree_variant = spree_variant
+      @shopify_variant = shopify_variant
+    end
+
+    def perform
+      shopify_variant.weight = spree_variant.weight
+      shopify_variant.weight_unit = 'oz'
+      shopify_variant.price = spree_variant.price
+      shopify_variant.sku = spree_variant.sku
+      shopify_variant.updated_at = spree_variant.updated_at
+      generate_options!
+
+      shopify_variant
+    end
+
+    private
+
+    attr_accessor :spree_variant, :shopify_variant
+
+    def generate_options!
+      shopify_variant.option1 = spree_variant.sku
```

Comment:
I think I'd try to hide this information in a separate method, maybe along the lines of:

```
def assign_variant_uniqueness_constraint(value)
  shopify_variant.option1 = value
end
```

I get several benefits from this:

1. The method is the sole owner of the knowledge that Shopify's uniqueness constraint is the value of `option1`.
2. It expresses the intent of the assignment. When `generate_options!` (or even `perform`) contains `assign_variant_uniqueness_constraint`, a first-time reader of the code has a better notion of what's happening.
3. I can expand this method with additional logic without making `generate_options!` more unwieldy. For example, I could perform some actual validation of the SKU's uniqueness.
4. It presents a more abstract interface, and makes the object a bit less Shopify-centric.